scripts/globals/combat/physical_utilities.lua

```diff
@@ -294,7 +326,178 @@ xi.combat.physical.calculateFTP = function(actor, wsTP1000, wsTP2000, wsTP3000)
     return fTP
 end
 
-xi.combat.physical.calculatePDIF = function(actor, additionalParamsHere)
+xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAttackMod, isCritical, applyLevelCorrection, tpIgnoresDefense, tpFactor)
+    local pDif = 0
+
+    -- Step 1: Attack / Defense Ratio
+    local baseRatio     = 0
+    local actorAttack   = math.floor(actor:getStat(xi.mod.ATT) * wsAttackMod)
+    local targetDefense = target:getStat(xi.mod.DEF)
+
+    -- Actor Attack modifiers.
+    if actor:hasStatusEffect(xi.effect.BUILDING_FLOURISH) then
+        local flourishEffect = actor:getStatusEffect(xi.effect.BUILDING_FLOURISH)
+
+        if flourishEffect:getPower() >= 2 then -- 2 or more Finishing Moves used.
+            actorAttack = actorAttack + 25 + flourishEffect:getSubPower()
+        end
+    end
+
+    -- Target Defense Modifiers.
+    local ignoreDefenseFactor = 1
+
+    if tpIgnoresDefense then
+        ignoreDefenseFactor = tpFactor
+    end
+
+    targetDefense = math.floor(targetDefense * ignoreDefenseFactor)
+
+    baseRatio = actorAttack / targetDefense
+
+    -- Apply cap to baseRatio.
+    local pDifBaseCap = pDifWeaponCapTable[weaponType][1] -- TODO: Does something affect this, like with final cap?
+
+    baseRatio = utils.clamp(baseRatio, 0, pDifBaseCap)
+
+    -- Step 2: cRatio (Level correction, corrected ratio) Zone based!
+    local levelDifFactor = 0
+
+    if applyLevelCorrection then
+        levelDifFactor = (target:getMainLvl() - actor:getMainLvl()) * 0.05
+    end
```

Comment:
If I understand this correctly the applyLevelCorrection is only applying to the specific zones listed in the level_correction.

https://www.bluegartr.com/threads/112776-Dev-Tracker-Findings-Posts-(NO-DISCUSSION)/page8
This says that the baseRation WAS 2.0 capped and it not virtually uncapped as of 2013 so this seems to be correct.

Notes for level correction:
+ Monsters do NOT suffer from negative level correction (Attacker < Defender)
+ Mobs, Avatars and pets only get bonuses, no penalties (if Attack > Defnder then cRatio + correction)
+ Avatars cap at 2.0 ratio for anything that checks tooWeak.
+ Total PDIF cap at the end for monsters is 4.0

Reply:
Correct. Since the inclussion of mobs over lvl 99, any zone that has this kind of mobs has level correction turned off, even if the target mob is not over lvl 99. In most cases, it doesnt affect lvl 75 servers, since those zones will be disabled.

Unfortunately, locus and apex mobs happen to be in legacy zones aswell, because of reasons. I'm sure very good ones too. /s

However, I have added a check for lvl 75 servers, so level correction always gets applied.

> Avatars and pets cap at 2.0 ratio for anything that checks tooWeak
Which ratio do you mean? The base ratio, or the final cap?

Reply:
> 

Oh sorry! cRatio should be capped for pets after level correction is applied. They added this in to make sure pets dont "overkill too much" - I can find the devblog this week - I should have it bookmarked somewhere I just need to dig it out